aten/src/ATen/DLConvertor.cpp

```diff
@@ -88,9 +88,9 @@ DLDataType getDLDataType(const Tensor& t) {
   return dtype;
 }
 
-static DLDevice getDLDevice(const Tensor& tensor, const int64_t& device_id) {
+static DLDevice getDLDevice(const Tensor& tensor, c10::DeviceIndex device_id) {
   DLDevice ctx;
-  ctx.device_id = device_id;
+  ctx.device_id = static_cast<int>(device_id);
```

Comment:
Explicitly  `int32_t` looks better?
`device_id` in DLDevice is a `int32_t`, see https://github.com/pytorch/pytorch/blob/f4cf25bb24be735b2502ae13f290017992c2fac8/aten/src/ATen/dlpack.h#L97-L105